packages/instrumentation-qdrant/src/instrumentation.ts

```diff
@@ -0,0 +1,293 @@
+  }: {
+    collectionName: string;
+    params:
+    | qdrant.Schemas["PointInsertOperations"]
+    | qdrant.Schemas["PointsSelector"]
+    | qdrant.Schemas["PointRequest"]
+    | qdrant.Schemas['SearchRequest']
+
+    methodName: string;
+  }): Span {
+    const span = this.tracer.startSpan(`qdrant.${collectionName}.${methodName}`, {
+      kind: SpanKind.CLIENT
+    });
+
+    try {
+      if (this._config.traceContent) {
+        if (methodName === "upsert") {
+          this._setUpsertAttributes(
+            span,
+            params as qdrant.Schemas["PointInsertOperations"],
+            methodName,
+          );
+        }
+        else if (methodName === "delete") {
+          this._setDeleteAttributes(span, params as qdrant.Schemas["PointsSelector"]);
+        }
+        else if (methodName === "retrieve") {
+          this._setRetrieveAttributes(span, params as qdrant.Schemas["PointRequest"]);
+        }
+
+        else if (methodName === "search") {
+          this._setSearchAttributes(span, params as qdrant.Schemas["SearchRequest"]);
+        }
+      }
+    } catch (e) {
+      this._diag.warn(e);
+      this._config.exceptionLogger?.(e);
+    }
+
+    return span;
+  }
+
+
+  private _setUpsertAttributes(
+    span: Span,
+    params: qdrant.Schemas["PointInsertOperations"],
+    method: "add" | "update" | "upsert",
+  ) {
+
+    if ("batch" in params) {
+      span.setAttribute(
+        `db.qdrant.${method}.points_count`,
+        JSON.stringify(params.batch.ids?.length),
+      );
+    }
+
+    else {
+      span.setAttribute(
+        `db.qdrant.${method}.points_count`,
+        JSON.stringify(params.points?.length),
+      );
+    }
+  }
+
+  private _setDeleteAttributes(span: Span, params: qdrant.Schemas["PointsSelector"]) {
+    if ("filter" in params) {
+      span.setAttribute(
+        "db.qdrant.delete.filter.must",
+        JSON.stringify(params.filter?.must),
+      );
+
+      span.setAttribute(
+        "db.qdrant.delete.filter.should",
+        JSON.stringify(params.filter?.should),
+      );
+
+      span.setAttribute(
+        "db.qdrant.delete.filter.must_not",
+        JSON.stringify(params.filter?.must_not),
+      );
+
+      span.setAttribute(
+        "db.qdrant.delete.filter.min_should",
+        JSON.stringify(params.filter?.min_should),
+      );
+    }
+
+    else {
+      span.setAttribute(
+        "db.qdrant.delete.point_ids",
+        JSON.stringify(params.points),
+      );
+    }
+  }
+
+  private _setRetrieveAttributes(span: Span, params: qdrant.Schemas["PointRequest"]) {
+    span.setAttribute(
+      "db.qdrant.retrieve.ids_count",
+      JSON.stringify(params.ids?.length),
+    );
+    span.setAttribute("db.qdrant.retrieve.with_payload", JSON.stringify(params.with_payload));
+    span.setAttribute("db.qdrant.retrieve.with_vector", JSON.stringify(params.with_vector));
+  }
+
+  private _setSearchAttributes(span: Span, params: qdrant.Schemas["SearchRequest"]) {
+    span.setAttribute(
```

Comment:
This should be sent as an event (see other instrumentations of Chroma / Pinecone)